### ISSUE TITLE
I need help with local readthedocs integration with Gitlab.

### ISSUE THREAD

**Issue by pilgrimage80** [readthedocs/readthedocs.org#9857]
Hello, I managed to install readthedocs locally on the gitlab server on redhat 8.7. I am now accessing the web interface via 127.0.0.1:8000. In short, here's what I want to do;

1. How can I make 127.0.0.0.1:8000 a domain name like gitlabreadthedocs.
2. When I open readthedocs from the web page, how can I open it with the user on sign with gitlab. Because when I enter this way I get the error " DoesNotExist at /account/gitlab/login.
3.I created gitlab incoming webhook from the integration section to import gitlab manual with readthedocs and I entered it into gitlab webhook but I get 404 error when I test it. 

In short, what I want to do is to create documents in readthedocs format automatically with gitlab, I want to integrate gitlab and readthedocs together, but I have no idea exactly how to do this. I need step by step support on this. 

Thank you.

**Comment by humitos** [comment]
Hi @pilgrimage80! At https://dev.readthedocs.io/en/latest/install.html#configuring-connected-accounts we have documented how to work with VCS providers. Please take a look at that guide to configure GitLab locally.

**Comment by pilgrimage80** [comment]
Hello, I will follow this document for gitlab, but especially how can I make the readthedocs server as myserverdocs:8000 or myserverdocs instead of 127.0.0.1:8000? I could not find where to make this setting in readthedocs. Do I need to make a change in a python file called settings.py? Or do I need to do it in 127.0.0.0.1:8000/admin or on djongo? Thanks for your help.

**Comment by pilgrimage80** [comment]
There are base.py dev.py test.py under the readthedocs settings folder, these are marked as production domains. if I change which file and what, I can access the readthedocs interface in the form of mydocs.org or mydocs instead of 127.0.0.1:8000 and integrate it with gitlab as mydocs? 
thankyou.

**Comment by agjohnson** [comment]
@pilgrimage80 to run an instance locally, you should not need to change the domain. The provided domain configured with our development instance is `https://devthedocs.org`, which will resolve to localhost.

If you are configuring your own internal instance and would like a different domain name, you can alter the domains used in your own settings file. Internal instances are beyond the scope of what we support however, the core team doesn't have the bandwidth to help with local configuration. You might be better off asking your question on Stack Overflow.

**Comment by pilgrimage80** [comment]
Hello, I installed with the guide in this link.
https://readthedocsorg.readthedocs.io/en/latest/install.html

In this link, I did not install it with the docker structure because it says su note in the guide.

Note
Take into account that this setup is intended for development purposes. We do not recommend to follow this guide to deploy an instance of Read the Docs for production.

https://dev.readthedocs.io/en/latest/install.html#configuring-connected-accounts

I have installed with an outdated document before and frankly these links are confusing as to which of your documents is the correct one and which one I should follow. If the correct document is the giant one, under what conditions should I consider your note?

**Comment by pilgrimage80** [comment]
What is the difference between these links?

**Comment by stsewd** [comment]
@pilgrimage80 https://docs.readthedocs.io/en/stable/ and https://dev.readthedocs.io/en/latest/ are the official documentations, other site you may find are from users just importing our project. But as mentioned, we don't offer support for custom installations, see https://docs.readthedocs.io/en/stable/open-source-philosophy.html.

**Comment by pilgrimage80** [comment]
Do I need to install docker redhat to run readthedocs? 
https://docs.docker.com/engine/install/rhel/

**Comment by agjohnson** [comment]
Closing this issue as it's a bit beyond what core team supports and doesn't look to be an issue with Read the Docs.
